File: server/src/main/java/com/willkamp/vial/implementation/VialConfig.java
package com.willkamp.vial.implementation;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import lombok.Getter;

@Getter
class VialConfig {
  private final int port;
  private final String address;
  private final boolean useTls;
  private final int maxContentLength;

  VialConfig() {
    Config config = ConfigFactory.load().getConfig("vial");
    port = config.getInt("port");
    address = config.getString("address");
    useTls = config.getBoolean("useTls");
    maxContentLength = config.getInt("maxContentLength");
  }
}

File: server/src/main/java/com/willkamp/vial/implementation/VialConfig.kt
package com.willkamp.vial.implementation

import com.typesafe.config.ConfigFactory

internal class VialConfig {
    val port: Int
    val address: String
    val maxContentLength: Int

    init {
        val config = ConfigFactory.load().getConfig("vial")
        port = config.getInt("port")
        address = config.getString("address")
        maxContentLength = config.getInt("maxContentLength")
    }
}
